epochStart/bootstrap/process.go

```diff
@@ -250,9 +250,13 @@ func (e *epochStartBootstrap) prepareEpochZero() (Parameters, error) {
 		return Parameters{}, err
 	}
 
+	shardIDToReturn := e.genesisShardCoordinator.SelfId()
+	if shardIDToReturn != e.destinationShardAsObserver && e.destinationShardAsObserver != core.DefaultShardIDAsObserver {
```

Comment:
same conditions duplicated in several places - make a function for it. This part should count only if node is not validator.